### PR TITLE
feat(cypress): add test case for incorrect and empty card types

### PR DESCRIPTION
**Pull request by kashif-m**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Test cases for #6512 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Adds test cases for card types.

## How did you test it?
Locally.

image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by Gnanasundari24 — DISMISSED** [review]

File: cypress-tests/cypress/e2e/PaymentUtils/Commons.js [hunk 1]
```diff
@@ -1229,6 +1229,79 @@ export const connectorDetails = {
...
+            line3: "Harrison Street",
+            city: "San Fransico",
+            state: "California",
+            zip: "94122",
+            country: "BR",
+            first_name: "john",
+            last_name: "doe",
+          },
+        },
+        payment_method_data: {
```
Comment (likhinbopanna) [inline cypress-tests/cypress/e2e/PaymentUtils/Commons.js:1252]: Can you pass the constant for the card?

File: cypress-tests/cypress/e2e/PaymentUtils/Commons.js [hunk 2]
```diff
@@ -1229,6 +1229,79 @@ export const connectorDetails = {
...
```
Comment (likhinbopanna) [inline cypress-tests/cypress/e2e/PaymentUtils/Commons.js:1288]: Here as well pass the constant ?

File: cypress-tests/cypress/e2e/PaymentUtils/Commons.js [hunk 3]
```diff
@@ -1229,6 +1229,79 @@ export const connectorDetails = {
...
       },
     },
+    IncorrectCardType: {
+      Request: {
+        currency: "USD",
+        payment_method: "card",
+        payment_method_type: "debit",
+        setup_future_usage: "off_session",
+        confirm: true,
+        billing: {
```
[the remark this answers is not in the document]
Comment (kashif-m) [reply cypress-tests/cypress/e2e/PaymentUtils/Commons.js:1239]: I don't see a constant for billing @pixincreate

File: cypress-tests/cypress/e2e/PaymentUtils/Commons.js [hunk 4]
```diff
@@ -98,6 +98,35 @@ const multiUseMandateData = {
...
+    line2: "Harrison Street",
+    line3: "Harrison Street",
+    city: "San Fransico",
+    state: "California",
+    zip: "94122",
+    country: "BR",
+    first_name: "john",
+    last_name: "doe",
+  },
+};
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/PaymentUtils/Commons.js:113]: if this is specific to a connector or a payment method, then i would recommend having country field empty and override it in respective fields wherever it is used. 

also, if i'm not asking too much, can you please also apply the same changes to rest of the file where ever `billing` exist and is not `null`.

that way, we can do something like:

```js
billing: {
  ...billingAddress,
  country: DE / GB / BR / IN / US
}
```

**Review by pixincreate — COMMENTED** [review]
looks good to me other wise

File: cypress-tests/cypress/e2e/PaymentUtils/Commons.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (pixincreate) [reply cypress-tests/cypress/e2e/PaymentUtils/Commons.js]: my bad, this should not have been done this way. please refer to this: https://github.com/juspay/hyperswitch/pull/6550#discussion_r1883625410
